### PR TITLE
【prim】add dropout composite rule

### PR DESCRIPTION
**Pull request by cxxly**
### PR types

New features
### PR changes

Others
### Describe

add dropout composite rule.

### CODE REVIEW

File: python/paddle/incubate/autograd/composite_rules.py [hunk 1]
```diff
@@ -20,6 +20,8 @@
 import functools
 import operator
 
+import paddle
```
Comment (cyber-pioneer) [inline python/paddle/incubate/autograd/composite_rules.py]: 尽量不要在这里直接引入paddle，防止出现引用循环依赖

File: python/paddle/fluid/tests/unittests/prim/composite_ops/test_composite_dropout.py [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+            paddle.seed(seed)
+            mp, sp = paddle.static.Program(), paddle.static.Program()
+            with paddle.static.program_guard(mp, sp):
+                input_ = paddle.static.data('x', shape=x.shape, dtype=x.dtype)
+                input_.stop_gradient = False
+                output = paddle.nn.functional.dropout(
+                    input_, p, training=(not is_test), mode=mode
+                )
+                if core._is_fwd_prim_enabled():
+                    paddle.incubate.autograd.to_prim(mp.blocks)
```
Comment (cyber-pioneer) [inline python/paddle/fluid/tests/unittests/prim/composite_ops/test_composite_dropout.py:157]: 补充op check，确认dropout算子被拆解

File: python/paddle/fluid/tests/unittests/prim/composite_ops/test_composite_dropout.py [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+                grad = paddle.static.gradients(output, input_)[0]
+            exe = paddle.static.Executor(self.place)
+            exe.run(sp)
+            return exe.run(mp, feed={input_.name: x}, fetch_list=[output, grad])
+
+        core._set_prim_forward_enabled(True)
```
Comment (cyber-pioneer) [inline python/paddle/fluid/tests/unittests/prim/composite_ops/test_composite_dropout.py]: 补充前反向都拆的case

File: python/paddle/incubate/autograd/composite_rules.py (hunk 1 above)
Comment (cxxly) [reply python/paddle/incubate/autograd/composite_rules.py]: done

File: python/paddle/fluid/tests/unittests/prim/composite_ops/test_composite_dropout.py (hunk 2 above)
Comment (cxxly) [reply python/paddle/fluid/tests/unittests/prim/composite_ops/test_composite_dropout.py:157]: done

File: python/paddle/fluid/tests/unittests/prim/composite_ops/test_composite_dropout.py (hunk 3 above)
Comment (cxxly) [reply python/paddle/fluid/tests/unittests/prim/composite_ops/test_composite_dropout.py]: done

**Review by cyber-pioneer — APPROVED** [review]

File: python/paddle/incubate/autograd/composite_rules.py [hunk 4]
```diff
@@ -149,3 +151,46 @@ def mean_composite(x, axis, keepdim):
         dtype=sum_x.dtype,
     )
     return divide(sum_x, norm)
+
+
+@REGISTER_COMPOSITE('dropout')
+def dropout_composite(x, seed_tensor, p, is_test, mode, seed, fix_seed):
```
Comment (jeff41404) [inline python/paddle/incubate/autograd/composite_rules.py:157]: The function signature should be consistent with dropout 2.x API

Comment (cxxly) [reply python/paddle/incubate/autograd/composite_rules.py:157]: 当前2.0API与legacy_ops.yaml定义中不一致，前向拆分是从op拆分到更细粒度op，所以签名暂无法与2.0API保持一致。后续legacy_ops迁移后，该部分会统一迁移

Comment (jeff41404) [reply python/paddle/incubate/autograd/composite_rules.py:157]: understand

**Review by jeff41404 — APPROVED** [review]
LGTM
